model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/UserCacheSession.java

```diff
@@ -240,6 +240,7 @@ static String getFederatedIdentityLinksCacheKey(String userId) {
     @Override
     public UserModel getUserByUsername(RealmModel realm, String username) {
         logger.tracev("getUserByUsername: {0}", username);
+        //TODO remove following line?
```

Comment:
Good question. I'd think we shouldn't change legacy behavior.

Reply:
No, we should keep it. It is the storage responsibility to deal with letter case. In the case of legacy storage, lettercase is intentionally ignored.